### PR TITLE
Add use_napalm parameter to NetBoxInventory2

### LINKED ISSUE #22: Give the ability to use Napalm driver as the platform rather than the actual platform name or slug
Currently if you want to integrate with the nornir_napalm plugin which seems to look at platform in host, you cannot have a differently named name or slug than the actual Napalm driver.

Seems like this will require another request be made to the Netbox API as the Napalm driver info is not passed in the devices endpoint (correct me if I'm wrong).

### PR DESCRIPTION
**Pull request by johanek**
Allow the use of napalm_driver value from the device platform to set the platform value

This could resolve #22 

I wasn't certain how to deal with the priority of use_platform_slug or use_napalm, so I defaulted to use_platform_slug "winning".

### CODE REVIEW

File: nornir_netbox/plugins/inventory/netbox.py [hunk 1]
```diff
@@ -211,13 +214,19 @@ def __init__(
...
         self.use_platform_slug = use_platform_slug
+        self.use_napalm = use_napalm
 
         self.session = requests.Session()
         self.session.headers.update({"Authorization": f"Token {nb_token}"})
         self.session.verify = ssl_verify
 
     def load(self) -> Inventory:
 
+        platforms: List[Dict[str, Any]] = []
```
Comment (wvandeun) [inline nornir_netbox/plugins/inventory/netbox.py:225]: We should fence this with a `if use_napalm`. This shouldn't be called if there is no intention for that.

**Comment by wvandeun** [comment]
Thx for the work on this.
Haven't had to time yet to look into this.

We could argue that both arguments can't be "True" at the same time and raise an exception?
I'll (try to) have a better look at this, this weekend.

File: nornir_netbox/plugins/inventory/netbox.py (hunk 1 above)
Comment (johanek) [reply nornir_netbox/plugins/inventory/netbox.py]: Good idea, updated

**Comment by johanek** [comment]
> We could argue that both arguments can't be "True" at the same time and raise an exception?

That sounds good to me - if it's what you think best I'll implement it and also update the README to explain the behaviour

File: nornir_netbox/plugins/inventory/netbox.py [hunk 2]
```diff
@@ -256,6 +266,10 @@ def load(self) -> Inventory:
 
             if isinstance(device["platform"], dict) and self.use_platform_slug:
                 platform = device["platform"].get("slug")
+            elif isinstance(device["platform"], dict) and self.use_napalm:
+                platform = [
+                    d for d in platforms if device["platform"]["slug"] == d["slug"]
```
Comment (wvandeun) [inline nornir_netbox/plugins/inventory/netbox.py:271]: Can you pick a better var name than `d`. `platform` makes a lot of sens to me.

File: README.md [hunk 3]
```diff
@@ -107,6 +107,8 @@ Arguments:
         (defaults to False)
     use_platform_slug: Use the NetBox platform slug for the platform attribute of a Host
         (defaults to False)
+    use_napalm: Use the Netbox platform napalm driver setting for the platform attribute
```
Comment (wvandeun) [inline README.md:110]: Can we rename the option to `use_platform_napalm_driver`. It better describes what the option exactly does

**Comment by wvandeun** [comment]
> > We could argue that both arguments can't be "True" at the same time and raise an exception?
> 
> That sounds good to me - if it's what you think best I'll implement it and also update the README to explain the behaviour

Yes, the should be "mutually exclusive". It doesn't make sense for them to be `True` at the same time and we should raise.
We will also need a testcase and the documentation needs to be updated.

**Comment by johanek** [comment]
Thanks for the review suggestions - I made the changes as requested.

The tests are failing on something unrelated - I also have some changes to update python versions and package dependencies which shows the tests passing, but I wanted to keep that change separate so as not to pollute this PR:

https://github.com/vdltech/nornir_netbox/compare/develop...vdltech:feature/fix-tests?expand=1

**Comment by wvandeun** [comment]
Can you rebase on develop branch? That should fix the tests.

**Comment by johanek** [comment]
> Can you rebase on develop branch? That should fix the tests.
> 
> The last piece I see missing is documentation:
> https://github.com/wvandeun/nornir_netbox/blob/develop/docs/usage.md#configuration-options

Hi, both are done 👍

**Comment by wvandeun** [comment]
LGTM! Thx

**Comment by johanek** [comment]
Nice one! Thank you for the project, it's really useful 👍

**Pull request merged**